modules/standard/Buffers.chpl

```diff
@@ -306,16 +308,65 @@ module Buffers {
      :arg error: (optional) capture an error that was encountered instead of
                   halting on error
    */
-  proc buffer.buffer(out error:syserr) {
+  proc buffer.init(out error:syserr) {
     this.home = here;
+    super.init();
     error = qbuffer_create(this._buf_internal);
   }
   pragma "no doc"
-  proc buffer.buffer() throws {
+  proc buffer.init() /*throws*/ {
     var error:syserr = ENOERR;
     this.home = here;
+    super.init();
     error = qbuffer_create(this._buf_internal);
-    if error then try ioerror(error, "in buffer constructor");
+    if error then try! ioerror(error, "in buffer constructor");
+  }
+  pragma "no doc"
+  proc buffer.init(x: buffer) {
+    if x.home == here {
+      this.home = here;
+      this._buf_internal = x._buf_internal;
+      super.init();
+      qbuffer_release(x._buf_internal);
+    } else {
+      var error: syserr = ENOERR;
+      this.init(error);
+      if error then halt("Got error on buffer_create");
```

Comment:
I might say "when initializing the type" instead, since "buffer_create" to me implies a call to a function named buffer_create.